aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -881,11 +894,24 @@ Tensor minimum(const Tensor& self, const Tensor& other) {
 
 // binary min, alias for minimum
 Tensor& min_out(Tensor& result, const Tensor& self, const Tensor& other) {
-  return at::minimum_out(result, self, other);
+    return at::minimum_out(result, self, other);
 }
 
 Tensor min(const Tensor& self, const Tensor& other) {
-  return at::minimum(self, other);
+    return at::minimum(self, other);
+}
+
+Tensor& fmin_out(Tensor& result, const Tensor& self, const Tensor& other) {
+    auto both_nan = self.isnan().logical_and_(other.isnan());
+    return at::minimum_out(result,
+                           at::where(self.isnan()  == both_nan, self, other.to(self.dtype())),
```

Comment:
This one allocates `self.isnan()` and `other.isnan()` twice. Should they be cached in a variable?

Reply:
Addressed! Thanks for correcting me.